ckanext/cioos_theme/plugin.py

```diff
@@ -443,6 +446,38 @@ def before_index(self, data_dict):
         # eov is multi select so it is a json list rather then a python list
         if(data_dict.get('eov')):
             data_dict['eov'] = load_json(data_dict['eov'])
+
+        # Index Source XML
+        # harvest object will be json if harvested from another ckan instance. TODO check it is xml
+        # try harvest object (xml)
+        h_object_id = data_dict.get('harvest_object_id', 'none')
+        context = {'model': model,
+                   'session': model.Session,
+                   'ignore_auth': True}
+
+        pkg_dict = {'id': h_object_id}
+
+        try:
+            harvest_object = toolkit.get_action('harvest_object_show')(context, pkg_dict)
+            content = harvest_object.get('content', '')
+            if content.startswith('<'):
+                data_dict['extras_harvest_document_content'] = harvest_object.get('content', '')
+            else:
+                raise NotFound
+        except NotFound:
+            log.warning('Unable to find harvest object "%s" or dose not contain valid xml '
+                        'referenced by dataset "%s". Trying xml url',
+                        pkg_dict['id'], data_dict['id'])
+
+            # try reading from xml url
+            xml_url = data_dict.get('xml_location_url')
+            try:
+                data_dict['extras_harvest_document_content'] = urllib2.urlopen(xml_url).read(100000)  # read only 100 000 chars
```

Comment:
You might want to add a check that the data retrieved from the url is actually xml, using an xml library. Just to make sure its not HTML we are indexing

Reply:
Also does cutting off the xml at 100kb work for SOLR indexing? eg the xml might not have valid structure if its chopped. Maybe solr doesnt care, if it does we could just discard large files

Reply:
Hey, thanks for the review! content of harvest_object is populated by the CKAN harvester or spatial harvester. as such it can only be JSON or XML. I could definitely use a library it seemed a bit heavy wait for this application. If you know of a good one I'm happy to try it though.

Cutting of the XML was just to prevent someone from killing our system with massive files. Solr is doing a text index on this field and all XML tags are stripped during indexing so it doesn't care what the format is.

Reply:
Just referring to line 475 here - correct me if I'm wrong but looks like the URL given could be anything not necessarily xml..  If you wanted to make sure downloaded content is xml you could use Python's built in xml library, eg:

```python
import xml.etree.ElementTree as ET
ET.fromstring('<somexml></somexml>')
```

Reply:
o right, good point. Thanks for catching that.